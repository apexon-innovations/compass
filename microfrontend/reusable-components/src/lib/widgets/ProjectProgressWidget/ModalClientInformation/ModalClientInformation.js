import React, { Component } from 'react'
import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import PropTypes from 'prop-types'
import { Scrollbars } from 'react-custom-scrollbars'
import { Row, Col } from 'react-bootstrap'
import { NotificationManager } from 'react-notifications'
import MembersSlider from '../MembersSlider/MembersSlider'
import { getColorBasedOnCode } from '../../../utils/commonFunction'
import { putApiCall } from '../../../utils/apiCall'
import { USER_DUMMY, IC_EDIT } from '../../../const/imgConst'
import Slider from 'react-slick'
import { Tooltip as ReactTooltip } from 'react-tooltip'
import Loader from '../../../components/Loader/Loader'
import FileUploader from '../../../components/FileUploader/FileUploader'
import style from './ModalClientInformation.module.scss'
import 'slick-carousel/slick/slick.css'

dayjs.extend(utc)

class ModalClientInformation extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isLoading: false,
    }
    this.inputFileRef = React.createRef()
  }

  getSettings(ownerLength, name) {
    return {
      infinite:
        name === 'Delivery Manager'
          ? ownerLength > 1
            ? true
            : false
          : ownerLength > 2
          ? true
          : false,
      speed: 800,
      focusOnSelect: false,
      dots: false,
      slidesToShow: name === 'Delivery Manager' ? 1 : 2,
      slidesToScroll: 1,
    }
  }

  renderMainInfo(name, value) {
    return (
      <div className={style.box}>
        <div className={style.label}>{name}</div>
        <div className={style.value}>{value}</div>
      </div>
    )
  }

  renderMangerDetails(name, data) {
    if (data) {
      const slides = []
      const toolTip = []
      for (let i = 0; i < data.length; i++) {
        slides.push(
          <div className={style.ownersBox} key={i} data-tip data-tooltip-id={`${name}-${i}`}>
            <div className={[style.userImg]}>
              <img src={data[i].dp ? data[i].dp : USER_DUMMY} title="" alt="" />
            </div>
            <div className={style.designation}>{name}</div>
            <div className={style.name}>{data[i].name}</div>
          </div>,
        )
        if (data[i].nestProjects) {
          toolTip.push(
            <ReactTooltip
              className="customThemeToolTip"
              key={`${name}-${i}`}
              id={`${name}-${i}`}
              border={true}
              delayHide={300}
              delayUpdate={300}
              borderColor="rgb(124, 180, 235)"
              arrowColor="rgba(2, 7, 27, 1)"
            >
              <h3>Managing Projects</h3>
              <ul>
                <Scrollbars autoHeight autoHeightMax={`115px`}>
                  {data[i].nestProjects.map((item, key) => {
                    return (
                      <React.Fragment key={`nest-${key}`}>
                        <li>{item}</li>
                      </React.Fragment>
                    )
                  })}
                </Scrollbars>
              </ul>
            </ReactTooltip>,
          )
        }
      }
      return (
        <React.Fragment>
          <div className={style.ownerSlider}>
            <div
              className={[style.sliderWrap, name === 'Delivery Manager' ? style.dm : ''].join(' ')}
            >
              <Slider {...this.getSettings(data.length, name)}>{slides}</Slider>
              {toolTip}
            </div>
          </div>
        </React.Fragment>
      )
    } else {
      return null
    }
  }

  onFileUpload(data) {
    const requestData = {
      fileName: data.name,
      file: data.base64,
    }
    this.setState({
      isLoading: true,
    })

    putApiCall(this.props.fileUploaderUrl, requestData)
      .then(responseData => {
        if (responseData.success) {
          NotificationManager.success('Project Logo Uploaded Successfully', '', 5000)
          window.dispatchEvent(new CustomEvent('updateProjectIcon', { detail: new Date() }), 1000)
        } else {
          NotificationManager.error(responseData.data.message, '', 5000)
        }
        this.setState({ isLoading: false })
      })
      .catch(() => {
        this.setState({ isLoading: false })
        NotificationManager.error('Something went wrong Please try after sometime', '', 5000)
      })
  }

  render() {
    const { data } = this.props
    const { isLoading } = this.state
    return (
      <div className={style.clientsInfo}>
        <Row>
          <Col md={5} sm={5} className="border-right pl-0">
            <Row>
              <Col md={5} sm={5} className="pl-0">
                <div
                  className={[
                    style.companyInfo,
                    style[
                      getColorBasedOnCode(data.healthMetrics ? data.healthMetrics.overall : 'NA')
                        .color
                    ],
                  ].join(' ')}
                >
                  <div className={[style.icon, data.iconLocation ? style.whiteBg : ''].join(' ')}>
                    {isLoading ? (
                      <Loader />
                    ) : (
                      <React.Fragment>
                        {data.iconLocation ? (
                          <img src={data.iconLocation} title={data.name} alt={data.name} />
                        ) : (
                          <h6>{data.initials}</h6>
                        )}
                        <div
                          className={style.editBox}
                          onClick={() => {
                            this.inputFileRef.current.fileInput.current.click()
                          }}
                        >
                          <img src={IC_EDIT} title="" alt="" />
                        </div>
                      </React.Fragment>
                    )}

                    <FileUploader
                      inputClassName="hide"
                      ref={this.inputFileRef}
                      onCompleted={e => {
                        this.onFileUpload(e)
                      }}
                      onError={e => {
                        NotificationManager.error(e.msg, '', 5000)
                      }}
                    />
                  </div>
                  <div className={style.ring}></div>
                </div>
              </Col>
              <Col md={7} sm={7} className="pl-0">
                <div className={style.companyDetails}>
                  {this.renderMainInfo('Client', data.clientName)}
                  {this.renderMainInfo('Project', data.name)}
                  {this.renderMainInfo('Billing Type', data.billingType)}
                  {this.renderMainInfo(
                    'Start Date',
                    dayjs(data.startDate)
                      .utc()
                      .format('MM-DD-YYYY'),
                  )}
                  {this.renderMainInfo(
                    'Completion Date',
                    dayjs(data.endDate)
                      .utc()
                      .format('MM-DD-YYYY'),
                  )}
                </div>
              </Col>
            </Row>
          </Col>
          <Col md={7} sm={7}>
            <div className={style.ownersInfo}>
              {this.renderMangerDetails('Delivery Manager', data.deliveryManager)}
              {this.renderMangerDetails('Project Manager', data.projectManager)}
            </div>
            <div className={style.membersInfo}>
              <Row>
                <Col md={3}>
                  <div className={style.membersCount}>
                    <div className={style.numberBox}>
                      <div className={style.number}>{data.resources.actual}</div>
                    </div>
                    <div className={style.label}>Members</div>
                  </div>
                </Col>
                <Col md={9}>
                  <div className={style.membersSlider}>
                    <MembersSlider data={data.teamMembers} />
                  </div>
                </Col>
              </Row>
            </div>
          </Col>
        </Row>
      </div>
    )
  }
}

ModalClientInformation.propTypes = {
  data: PropTypes.object,
  fileUploaderUrl: PropTypes.string,
}

export default ModalClientInformation
