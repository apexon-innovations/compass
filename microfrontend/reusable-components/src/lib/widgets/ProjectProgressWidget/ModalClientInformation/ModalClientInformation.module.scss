@import '../../../theme/theme.scss';

.clientsInfo {
  width: 100%;
  padding: 0 15px;
}

.companyInfo {
  width: 100%;
  position: relative;
  margin: 10px 0 0 0;

  &.green {
    .icon {
      background: $themeGreenMain;
      h6 {
        color: $whiteColor;
      }
    }
    .ring {
      border-color: $themeGreenMain;
    }
  }
  &.red {
    .icon {
      background: $themeRedMain;
      h6 {
        color: $whiteColor;
      }
    }
    .ring {
      border-color: $themeRedMain;
    }
  }
  &.amber {
    .icon {
      background: $themeAmberMain;
      h6 {
        color: $whiteColor;
      }
    }
    .ring {
      border-color: $themeAmberMain;
    }
  }

  .icon {
    width: 80px;
    height: 80px;
    @include border-radius(50%);
    background: $whiteColor;
    @include flexbox();
    @include align-items(center);
    @include justify-content(center);
    position: relative;
    overflow: hidden;
    z-index: 10;
    margin: 0 auto;
    cursor: pointer;
    &.whiteBg {
      background: $whiteColor;
    }

    &:hover {
      .editBox {
        bottom: 0;
      }
    }

    .editBox {
      width: 100%;
      height: 30px;
      background: $themeBlueMain;
      @include position(absolute, unset, unset, -100%, 0);
      text-align: center;
      padding: 5px 0;
      cursor: pointer;
      @include transition(bottom, 0.4s, ease);

      img {
        max-width: 16px;
        max-height: 16px;
      }
    }

    h6 {
      font-size: 14px;
      font-weight: 600;
      text-decoration: none !important;
      letter-spacing: -1px;
      margin: 0;
      color: $grey33;
    }

    img {
      max-width: 50px;
      max-height: 50px;
    }
  }
  .ring {
    width: 95px;
    height: 95px;
    @include position(absolute, 50%, unset, unset, 50%);
    transform: translate(-50%, -50%);
    z-index: 0;
    @include border-radius(50%);
    border: 3px solid;
  }
}

.companyDetails {
  width: 100%;
  text-align: left;

  .box {
    width: 100%;
    @include flexbox();
    @include flex-direction(column);
    margin: 0 0 10px 0;

    .label {
      width: 100%;
      font-size: 14px;
      @include truncate(100%);
      color: $whiteColor;
    }
    .value {
      width: 100%;
      font-size: 16px;
      color: $themeBlueMain;
    }
  }
}

.ownersInfo {
  width: 100%;
  @include flexbox();
  @include justify-content(space-around);
  margin: 20px 0 30px 0;

  .ownersBox {
    text-align: left;

    .userImg {
      width: 80px;
      height: 80px;
      overflow: hidden;
      margin: 0 0 10px 0;
      @include border-radius(10px);
      border: 1px solid $whiteColor;

      img {
        max-width: 100%;
        max-height: 100%;
      }
    }
    .designation {
      font-size: 14px;
      color: $whiteColor;
    }
    .name {
      font-size: 18px;
      color: $themeAmberMain;
    }
  }
}

.membersInfo {
  width: 100%;

  .membersCount {
    @include flexbox();
    @include flex-direction(column);
    @include justify-content(center);
    text-align: center;

    .numberBox {
      margin: 0 0 10px 0;

      .number {
        max-width: 60px;
        width: 100%;
        height: 60px;
        position: relative;
        @include border-radius(50%);
        text-align: center;
        @include flexbox();
        @include align-items(center);
        @include justify-content(center);
        border: 1px dashed $themeBlueMain;
        color: $themeBlueMain;
        font-size: 24px;
        font-weight: 600;
        margin: 0 auto;
      }
    }
    .label {
      font-size: 16px;
    }
  }
}

.membersSlider {
  width: 100%;
  position: relative;
  margin: 15px 0 0 0;
}

.ownerSlider {
  width: 100%;
  position: relative;

  :global(.slick-track) {
    min-width: 200px;
    height: auto !important;
  }

  :global(.slick-slide) {
    width: auto;
    min-width: 110px;
  }
  :global(.slick-list) {
    width: calc(100% - 50px);
    left: 25px;
  }
  :global(.slick-prev),
  :global(.slick-next) {
    font-size: 0;
    line-height: 0;
    position: absolute;
    top: 50%;
    display: block;
    width: 20px;
    height: 20px;
    padding: 0;
    transform: translate(0, -50%);
    cursor: pointer;
    color: transparent;
    border: none;
    outline: none;
    background: transparent;

    &:before {
      content: '';
      width: 20px;
      height: 20px;
      @include position(absolute, 0, unset, unset, 0);
    }
  }
  :global(.slick-prev) {
    left: 0;
    &:before {
      background: url(../../../images/ic_arrow_left.svg) no-repeat 50% 50%;
    }
  }
  :global(.slick-next) {
    right: 0;
    &:before {
      background: url(../../../images/ic_arrow_right.svg) no-repeat 50% 50%;
    }
  }
}

.sliderWrap {
  max-width: 330px;
  position: relative;
  z-index: 10;

  &.dm {
    max-width: 170px;
  }
}

.sliderDiv {
  margin: 0 auto;
  max-width: 45px;
}
