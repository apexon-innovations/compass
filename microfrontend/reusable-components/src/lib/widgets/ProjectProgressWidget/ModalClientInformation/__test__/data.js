const data = {
  data: {
    id: '5ee31002d48d514e801cb10b',
    name: 'Compass',
    clientName: 'Infostretch',
    initials: 'C',
    category: 'All (Dev & QA & Support)',
    iconLocation:
      'http://sampleicon.com/image.png',
    billingType: 'Internal',
    startDate: 'Wed Jan 01 00:00:00 UTC 2020',
    endDate: 'Thu Jan 02 00:00:00 UTC 2020',
    resources: {
      actual: 10,
      sow: 10,
    },
    healthMetrics: {
      overall: 'R',
      periodicReview: 'G',
      scope: 'A',
      deliverables: 'A',
      customer: 'G',
      schedule: 'R',
      financials: 'G',
      people: 'G',
      compliance: 'G',
      startUp: 'G',
      process: 'G',
      planning: 'G',
    },
    projectManager: [],
    deliveryManager: [
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
        nestProjects: ['Infostretch - DES'],
      },
    ],
    accountManager: [
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
        nestProjects: ['Infostretch - DES'],
      },
    ],
    teamMembers: [
      {
        email: 'Pradeep.shrivastava@infostretch.com',
        name: 'Pradeep Shrivastava',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933',
      },
      {
        email: 'vijay.solanki@infostretch.com',
        name: 'Vijay Solanki',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621',
      },
      {
        email: 'suraj.bhosale@infostretch.com',
        name: 'Suraj Bhosale',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950',
      },
      {
        email: 'yatin.bhavsar@infostretch.com',
        name: 'Yatin Bhavsar',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973',
      },
      {
        email: 'vishal.kumar@infostretch.com',
        name: 'Vishal kumar',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930',
      },
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
      },
      {
        email: 'namra.porecha@infostretch.com',
        name: 'Namra Porecha',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375',
      },
      {
        email: 'purushottam.Khandebharad@infostretch.com',
        name: 'Purushottam Khandebharad',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437',
      },
      {
        email: 'amit.ghadge@infostretch.com',
        name: 'Amit Ghadge',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269',
      },
    ],
  },
  fileUploaderUrl:
    'https://compass-dev-api.infostretch.com/psr-service/project/5ee31002d48d514e801cb10b/logo',
}

export default data
