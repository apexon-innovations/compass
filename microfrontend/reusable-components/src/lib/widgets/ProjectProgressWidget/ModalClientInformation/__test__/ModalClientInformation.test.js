/* eslint-disable */
import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import { sleep } from '../../../../../__tests__/commonTestFunction'
import ModalClientInformation from '../ModalClientInformation'
import data from './data'
const props = { ...data }

describe(`ModalClientInformation`, () => {
  let container = null

  beforeEach(() => {
    // set up a DOM element as a render target
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container)
    container.remove()
    container = null
  })

  it(`Snapshot`, async () => {
    act(() => {
      render(<ModalClientInformation {...props} />, container)
    })
    await act(() => sleep(500))
    expect(container).toMatchSnapshot()
  })
})
