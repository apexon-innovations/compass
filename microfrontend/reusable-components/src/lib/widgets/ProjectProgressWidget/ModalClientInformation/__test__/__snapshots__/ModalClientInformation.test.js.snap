// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalClientInformation Snapshot 1`] = `
<div>
  <div>
    <div
      class="row"
    >
      <div
        class="border-right pl-0 col-md-5 col-sm-5"
      >
        <div
          class="row"
        >
          <div
            class="pl-0 col-md-5 col-sm-5"
          >
            <div
              class=" "
            >
              <div
                class=" "
              >
                <img
                  alt="Compass"
                  src="http://sampleicon.com/image.png"
                  title="Compass"
                />
                <div>
                  <img
                    alt=""
                    src="[object Object]"
                    title=""
                  />
                </div>
                <input
                  accept="image/png, image/jpeg, image/svg+xml"
                  class="hide"
                  name="fileUploader"
                  type="file"
                />
              </div>
              <div />
            </div>
          </div>
          <div
            class="pl-0 col-md-7 col-sm-7"
          >
            <div>
              <div>
                <div>
                  Client
                </div>
                <div>
                  Infostretch
                </div>
              </div>
              <div>
                <div>
                  Project
                </div>
                <div>
                  Compass
                </div>
              </div>
              <div>
                <div>
                  Billing Type
                </div>
                <div>
                  Internal
                </div>
              </div>
              <div>
                <div>
                  Start Date
                </div>
                <div>
                  01-01-2020
                </div>
              </div>
              <div>
                <div>
                  Completion Date
                </div>
                <div>
                  01-02-2020
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-7 col-sm-7"
      >
        <div>
          <div>
            <div
              class=" "
            >
              <div
                class="slick-slider slick-initialized"
              >
                <div
                  class="slick-list"
                >
                  <div
                    class="slick-track"
                    style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                  >
                    <div
                      aria-hidden="false"
                      class="slick-slide slick-active slick-current"
                      data-index="0"
                      style="outline: none; width: 0px;"
                      tabindex="-1"
                    >
                      <div>
                        <div
                          data-tip="true"
                          data-tooltip-id="Delivery Manager-0"
                          style="width: 100%; display: inline-block;"
                          tabindex="-1"
                        >
                          <div
                            class=""
                          >
                            <img
                              alt=""
                              src="https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182"
                              title=""
                            />
                          </div>
                          <div>
                            Delivery Manager
                          </div>
                          <div>
                            Gunjan Shah
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class=" "
            >
              <div
                class="slick-slider slick-initialized"
              >
                <div
                  class="slick-list"
                >
                  <div
                    class="slick-track"
                    style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="row"
          >
            <div
              class="col-md-3"
            >
              <div>
                <div>
                  <div>
                    10
                  </div>
                </div>
                <div>
                  Members
                </div>
              </div>
            </div>
            <div
              class="col-md-9"
            >
              <div>
                <div>
                  <div>
                    <div
                      class="slick-slider slick-initialized"
                      dir="ltr"
                    >
                      <button
                        class="slick-arrow slick-prev"
                        data-role="none"
                        style="display: block;"
                        type="button"
                      >
                         
                        Previous
                      </button>
                      <div
                        class="slick-list"
                      >
                        <div
                          class="slick-track"
                          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                        >
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="-4"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="gunjan.shah@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182"
                                    title="Gunjan Shah"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="-3"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="namra.porecha@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375"
                                    title="Namra Porecha"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="-2"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="purushottam.Khandebharad@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437"
                                    title="Purushottam Khandebharad"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="-1"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="amit.ghadge@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269"
                                    title="Amit Ghadge"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="false"
                            class="slick-slide slick-active slick-current"
                            data-index="0"
                            style="outline: none; width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="Pradeep.shrivastava@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933"
                                    title="Pradeep Shrivastava"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="false"
                            class="slick-slide slick-active"
                            data-index="1"
                            style="outline: none; width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="vijay.solanki@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621"
                                    title="Vijay Solanki"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="false"
                            class="slick-slide slick-active"
                            data-index="2"
                            style="outline: none; width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="suraj.bhosale@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950"
                                    title="Suraj Bhosale"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="false"
                            class="slick-slide slick-active"
                            data-index="3"
                            style="outline: none; width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="yatin.bhavsar@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973"
                                    title="Yatin Bhavsar"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide"
                            data-index="4"
                            style="outline: none; width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="vishal.kumar@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930"
                                    title="Vishal kumar"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide"
                            data-index="5"
                            style="outline: none; width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="gunjan.shah@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182"
                                    title="Gunjan Shah"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide"
                            data-index="6"
                            style="outline: none; width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="namra.porecha@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375"
                                    title="Namra Porecha"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide"
                            data-index="7"
                            style="outline: none; width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="purushottam.Khandebharad@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437"
                                    title="Purushottam Khandebharad"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide"
                            data-index="8"
                            style="outline: none; width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="amit.ghadge@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269"
                                    title="Amit Ghadge"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="9"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="Pradeep.shrivastava@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933"
                                    title="Pradeep Shrivastava"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="10"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="vijay.solanki@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621"
                                    title="Vijay Solanki"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="11"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="suraj.bhosale@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950"
                                    title="Suraj Bhosale"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="12"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="yatin.bhavsar@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973"
                                    title="Yatin Bhavsar"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="13"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="vishal.kumar@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930"
                                    title="Vishal kumar"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="14"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="gunjan.shah@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182"
                                    title="Gunjan Shah"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="15"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="namra.porecha@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375"
                                    title="Namra Porecha"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="16"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="purushottam.Khandebharad@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437"
                                    title="Purushottam Khandebharad"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            class="slick-slide slick-cloned"
                            data-index="17"
                            style="width: 0px;"
                            tabindex="-1"
                          >
                            <div>
                              <div
                                style="width: 100%; display: inline-block;"
                                tabindex="-1"
                              >
                                <div>
                                  <img
                                    alt="amit.ghadge@infostretch.com"
                                    src="https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269"
                                    title="Amit Ghadge"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <button
                        class="slick-arrow slick-next"
                        data-role="none"
                        style="display: block;"
                        type="button"
                      >
                         
                        Next
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
